api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -863,18 +863,29 @@ type AgentNodePoolPlatform struct {
 }
 
 type AzureNodePoolPlatform struct {
+	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	//
+	// +kubebuilder:validation:Required
+	// +required
 	VMSize string `json:"vmsize"`
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used:
-	// subscription/$subscriptionID/resourceGroups/$resourceGroupName/providers/Microsoft.Compute/images/rhcos.x86_64.vhd
+
+	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used and
+	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
+	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
+	// Hosted Cluster specification respecitvely, hcluster.Spec.Platform.Azure.SubscriptionID and
+	// hostedcluster.Spec.Platform.Azure.ResourceGroupName.
+	//
 	// +optional
 	ImageID string `json:"imageID,omitempty"`
```

Comment:
Similar comment as: https://github.com/openshift/hypershift/pull/4004/files#r1595143214

Reply:
> 	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the

Is there some specific reason the resource group name must be the same one as in the Hosted Cluster?

Reply:
It's just how it was previously coded by an engineer that left the team. This is most likely going to change in the future too once we determine where the RHCOS images are going to be stored.

Reply:
Understood :+1: thanks